### PR TITLE
eliminate global static GCS_MAVLINK::send_home_all

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by OXINARF** [comment]
LGTM, but I'll make two notes before merging:

- behavior changes as it will no longer only send home to active channels
- a new *chan* method wasn't needed, only making the current one be const (instead of the new one which is const and returns a const reference).

**Comment by peterbarker** [comment]
@OXINARF a const function returning a reference must return a const reference, which is unfortunate if you would like to fiddle with the returned value....

I've changed the GCS_MAVLink patch to take into account the active channel mask.  That mask is going to die in future, replaced with a `.active` or `.active()` methodcall.  Similarly with the streaming channel mask.

**Comment by amilcarlucas** [comment]
Sorry, I do not get the advantage of this PR. Can you elaborate Please ?

**Comment by magicrub** [comment]
@peterbarker has had this task in the queue for a long time. We've been wanting to change the whole GCS messaging system to use a singleton instead of all the libraries and vehicles  reaching back and forth over the GCS_MAVLink:: const reference which is a bit ugly. He has a series of PRs lined up to totally remove the GCS const refs which were held off to make sure it doesn't conflict with the copter release

**Comment by OXINARF** [comment]
> a const function returning a reference must return a const reference, which is unfortunate if you would like to fiddle with the returned value....

@peterbarker Ah you are right and it makes sense, you can't mark the method const if you give the caller the chance to change the internal state of the object. Learning every day! 🙂

@magicrub Why was this marked as DevCallTopic? This looks ready to go in, there is no change in behavior I can see.

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -1517,26 +1517,26 @@ void GCS_MAVLINK::send_home(const Location &home) const
...
+                continue;                            \
+            }                                        \
+            if (!(GCS_MAVLINK::active_channel_mask() & (chan(i).get_chan()-MAVLINK_COMM_0))) { \
+                continue;                            \
+            }                                        \
+            chan(i).methodcall;                      \
+        }                                            \
+    } while (0);
+
+void GCS::send_home(const Location &home) const
```
Comment (OXINARF) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: Given we already have a GCS.cpp file should this go there? I think it would be better, easier to find.

Comment (peterbarker) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: Yep!  Hadn't been created when I originally made this PR, I think.

Moved!

**Comment by OXINARF** [comment]
Merged, thanks! I've changed the commit message of the last commit here by adding "*to GCS_Dummy*".

**Pull request closed without merging**